application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java

```diff
@@ -79,11 +80,10 @@ private void handleGetMessage(String message, MessageReceivedEvent event) {
         String key = data[1];
 
         try {
-            Optional<StorageRecord> foundValue = database.read(ctx -> {
-                return ctx.selectFrom(Storage.STORAGE)
-                          .where(Storage.STORAGE.KEY.eq(key))
-                          .stream()
-                          .findFirst();
+            Optional<StorageRecord> foundValue = database.read(context -> {
+                try (var query = context.selectFrom(Storage.STORAGE)) {
+                    return query.where(Storage.STORAGE.KEY.eq(key)).stream().findFirst();
+                }
```

Comment:
```suggestion
                return Optional.ofNullable(context.selectFrom(Storage.STORAGE)
                                                  .where(Storage.STORAGE.KEY.eq(key))
                                                  .fetchOne());
```
Maybe? That way we can avoid the annoying try there.

Reply:
The linter says that `context.selectFrom(Storage.STORAGE)` opens a resource that has to be closed. Does `fetchOne` avoid that?

Reply:
Yes https://blog.jooq.org/tag/autocloseable/